tokio/src/runtime/context.rs

```diff
@@ -0,0 +1,114 @@
+//! Global runtime context
+#![allow(dead_code)]
+
+use crate::runtime::{time, Spawner};
+
+use std::cell::RefCell;
+
+thread_local! {
+    static CONTEXT: RefCell<Option<ThreadContext>> = RefCell::new(None)
+}
+
+/// ThreadContext makes Runtime context accessible to each Runtime thread.
+#[derive(Debug, Clone)]
+pub(crate) struct ThreadContext {
+    /// Handles to the executor.
+    spawner: Spawner,
+
+    /// Handles to the I/O drivers
+    io_handle: crate::runtime::io::Handle,
+
+    /// Handles to the time drivers
+    time_handle: crate::runtime::time::Handle,
+
+    /// Source of `Instant::now()`
+    clock: Option<crate::runtime::time::Clock>,
+}
+
+impl ThreadContext {
+    pub(crate) fn new() -> Self {
+        ThreadContext {
+            spawner: Spawner::Shell,
+            io_handle: None,
```

Comment:
I don't think this should compile right? https://github.com/tokio-rs/tokio/blob/master/tokio/src/runtime/io.rs#L57

Reply:
Hmm I think Github might be messed up, these are handled on my branch
https://github.com/gardnervickers/tokio/blob/thread-locals/tokio/src/runtime/context.rs#L30-L32